bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableDataSettingsFactory.java

```diff
@@ -0,0 +1,174 @@
+import com.google.cloud.bigtable.config.CredentialOptions.CredentialType;
+import com.google.cloud.bigtable.config.Logger;
+import com.google.cloud.bigtable.config.RetryOptions;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings.Builder;
+import com.google.cloud.bigtable.data.v2.models.InstanceName;
+import com.google.cloud.bigtable.data.v2.stub.BigtableStubSettings;
+
+/**
+ * Static methods to convert an instance of {@link Configuration} or {@link BigtableOptions} to a
+ * {@link BigtableDataSettings} instance.
+ */
+public class BigtableDataSettingsFactory {
+  /** Constant <code>LOG</code> */
+  private static final Logger LOG = new Logger(BigtableDataSettingsFactory.class);
+
+  /**
+   * <p>
+   * fromBigtableOptions.
+   * </p>
+   * @param configuration a {@link Configuration} object.
+   * @return a {@link BigtableDataSettings} object.
+   * @throws IOException if any.
+   */
+  public static BigtableDataSettings fromBigtableOptions(final BigtableOptions options)
+      throws IOException, GeneralSecurityException {
+    BigtableDataSettings.Builder builder = BigtableDataSettings.newBuilder();
+
+    InstanceName instanceName = InstanceName.newBuilder().setProject(options.getProjectId())
+        .setInstance(options.getInstanceId()).build();
+    builder.setInstanceName(instanceName);
+    builder.setAppProfileId(options.getAppProfileId());
+
+    LOG.debug("endpoint host %s.", options.getDataHost());
+    LOG.debug("endpoint host %s.", options.getPort());
+    builder.setEndpoint(options.getDataHost() + ":" + options.getPort());
+
+    createCredentialProvider(builder, options.getCredentialOptions());
+
+    setBulkOptions(builder, options.getBulkOptions());
+
+    RetryOptions retryOptions = options.getRetryOptions();
+    if (retryOptions.enableRetries()) {
+      // Creating retrySetting from retryOptions as well callOptionsConfig
+      RetrySettings retrySettings = RetrySettings.newBuilder()
+          .setInitialRetryDelay(ofMillis(retryOptions.getInitialBackoffMillis()))
+          .setRetryDelayMultiplier(retryOptions.getBackoffMultiplier())
+          .setMaxRetryDelay(ofMillis(retryOptions.getMaxElapsedBackoffMillis()))
+          .setMaxAttempts(retryOptions.getMaxScanTimeoutRetries())
+          .setInitialRpcTimeout(
+            ofMillis(options.getCallOptionsConfig().getShortRpcTimeoutMs()))
+          .setMaxRpcTimeout(ofMillis(options.getCallOptionsConfig().getLongRpcTimeoutMs()))
+          .build();
+      //TODO rahulkql:  Is this correct??, as these settings are for BigtableOptions 
+      //whole but here we are just setting it for bulkMutation.
+      builder.bulkMutationsSettings().setRetrySettings(retrySettings);
+      builder.checkAndMutateRowSettings().setRetrySettings(retrySettings);
+      builder.readModifyWriteRowSettings().setRetrySettings(retrySettings);
+      builder.readRowsSettings().setRetrySettings(retrySettings);
```

Comment:
* readRows should use the long timeout from CallOptionsConfig for totalTimeout
* readRows should use the ReadPartialRowTimeoutMillis as the rpc timeout